nrich-excel/src/main/java/net/croz/nrich/excel/generator/PoiExcelReportGenerator.java

```diff
@@ -54,24 +55,30 @@ public class PoiExcelReportGenerator implements ExcelReportGenerator {
 
     private final SXSSFWorkbook workbook;
 
-    private final Sheet sheet;
+    private final SXSSFSheet sheet;
 
     private final CreationHelper creationHelper;
 
     private final Map<Integer, CellStyle> cellStyleMap;
 
     private final Map<Class<?>, CellStyle> defaultStyleMap;
 
+    private final boolean autoSizeColumns;
+
     private int currentRowNumber;
 
     private boolean templateOpen = true;
 
     public PoiExcelReportGenerator(List<CellValueConverter> cellValueConverterList, OutputStream outputStream, InputStream template, List<TemplateVariable> templateVariableList,
-                                   List<TypeDataFormat> typeDataFormatList, List<ColumnDataFormat> columnDataFormatList, int startIndex) {
+                                   List<TypeDataFormat> typeDataFormatList, List<ColumnDataFormat> columnDataFormatList, int startIndex, boolean autoSizeColumns) {
         this.cellValueConverterList = cellValueConverterList;
         this.outputStream = outputStream;
         this.workbook = initializeWorkBookWithTemplate(template, templateVariableList);
         this.sheet = workbook.getSheetAt(0);
+        this.autoSizeColumns = autoSizeColumns;
```

Comment:
Maybe move this to the end to improve readability and to match the order of constructor parameters.